AsyncDisplayKit/ASButtonNode.mm

```diff
@@ -480,7 +481,11 @@ - (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
   }
   
   if (_titleNode.attributedText.length > 0) {
-    [children addObject:_titleNode];
+    if (_imageAlignment == ASButtonNodeImageAlignmentBeginning) {
+      [children addObject:_titleNode];
+    } else {
+      [children insertObject:_titleNode atIndex:0];
```

Comment:
I agree. In this case, though, earlier in this method we create the `children` array with capacity set to 2. Although it could still grow, but this spec has been setup to have only two children. What this does, only, is to add the title at the beginning in case the image is not.